collector/network_route_linux.go

```diff
@@ -76,15 +81,56 @@ func (n networkRouteCollector) Update(ch chan<- prometheus.Metric) error {
 		return fmt.Errorf("couldn't get routes: %w", err)
 	}
 
+	routeTableIDName, err := routeTableIDToString()
+	if err != nil {
+		return fmt.Errorf("couldn't get route table names: %w", err)
+	}
+
 	for _, route := range routes {
-		if route.Type != unix.RTA_DST {
-			continue
+		if route.Type == unix.RTN_BLACKHOLE || route.Type == unix.RTN_UNREACHABLE {
```

Comment:
Can we convert this into a switch statement or use early returns? Generally would be nice to avoid these deeply nested conditions.

Reply:
Thank you for the feedback. I've tried to clean it up.